File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java

import java.util.List;

class ResistorColorDuo {
    int value(String[] colors) {
        StringBuilder resistorValue = new StringBuilder();

        int count = 0;

        for (String color : colors) {
            count++;
            if (count > 2) {
                break;
            }
            int value = colorCode(color);

            if (value != -1) {
                resistorValue.append(value);
            }
        }

        return Integer.parseInt(resistorValue.toString(), 10);
    }

    private int colorCode(String color) {
        return colors().indexOf(color);
    }

    private List<String> colors() {
        return List.of(
                "black",
                "brown",
                "red",
                "orange",
                "yellow",
                "green",
                "blue",
                "violet",
                "grey",
                "white");
    }
}

File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
public static class ResistorColorDuo
{

    private const int ResistorLength = 2;
    
    public static int Value(string[] colors)
    {
        if (colors.Length < 2)
        {
            throw new ArgumentException("Not enough bands to determine resistance", nameof(colors) );
        }

        return colors
            .Take(ResistorLength)
            .Select((color, index) =>
            {
                var multiplier = (int)Math.Pow(10, ResistorLength - index - 1);
                var resistance = ColorToResistance(color);
                
                return multiplier * resistance;
            })
            .Sum();
    }

    private static int ColorToResistance(string color) => color.ToLower() switch
    {
        "black" => 0,
        "brown" => 1,
        "red" => 2,
        "orange" => 3,
        "yellow" => 4,
        "green" => 5,
        "blue" => 6,
        "violet" => 7,
        "grey" => 8,
        "white" => 9,
        _ => throw new ArgumentOutOfRangeException(nameof(color), color, null)
    };
    
}
